File: frontend/web/src/api/products/getProducts.ts
import axiosInstance from "../../utils/AxiosInstance.ts"

const getProducts = async () => {
    const token = localStorage.getItem("token");

    const { data } = await axiosInstance.get("/api/products", {
        headers: { Authorization: `Bearer ${token}` }
    });

    return data.products;
}

export default getProducts;

File: frontend/desktop/python-client/PyQt5App/api/products/getProducts.py
import requests
from PyQt5App.utils.Address import address
from PyQt5App.utils.sessionManager import SessionManager

class GetProducts:
    def __init__(self):
        self.address = address()
        self.sessionManager = SessionManager()
        self.token = self.sessionManager.get_token()

    def getProducts(self):
        response = requests.get(
            f"{self.address}/api/products",
            headers={
                "Authorization": f"{self.token}",
            }
        )

        data = response.json()
        products = data["products"]

        return products